WeakAuras/RegionTypes/AuraBar.lua

```diff
@@ -508,20 +551,43 @@ local barPrototype = {
 
   -- Set bar color
   ["SetForegroundColor"] = function(self, r, g, b, a)
-    self.fg:SetVertexColor(r, g, b, a);
+    if self.gradientFG then
```

Comment:
And here, you'd only store the foreground colors, and then call UpdateForegroundColor()

(And similar for background color)